rskj-core/src/test/java/co/rsk/peg/P2shErpFederationTest.java

```diff
@@ -10,37 +16,55 @@
 import co.rsk.config.BridgeMainNetConstants;
 import co.rsk.config.BridgeRegTestConstants;
 import co.rsk.config.BridgeTestNetConstants;
+import co.rsk.peg.bitcoin.BitcoinTestUtils;
+import java.time.Instant;
+import java.time.ZonedDateTime;
+import java.util.Arrays;
+import java.util.List;
 import java.util.stream.Collectors;
+import java.util.stream.Stream;
 import org.bouncycastle.util.encoders.Hex;
 import org.ethereum.config.blockchain.upgrades.ActivationConfig;
 import org.ethereum.config.blockchain.upgrades.ActivationConfigsForTest;
 import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
 
-import java.time.Instant;
-import java.time.ZonedDateTime;
-import java.util.Arrays;
-import java.util.List;
+class P2shErpFederationTest {
 
-import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.mockito.Mockito.mock;
+    @ParameterizedTest
+    @MethodSource("getRedeemScriptArgsProvider")
+    void getRedeemScript(BridgeConstants bridgeConstants) {
```

Comment:
What about testing that redeem scripts are not equal and passing an unordered list of keys?

Reply:
Not sure I got that. The keys are sorted in bitcoinj-thin when creating the redeem script anyway